### PR TITLE
added search limits

### PR DESCRIPTION
**Pull request by DFSko**
resolves happy-travel/agent-app-project#993
resolves happy-travel/agent-app-project#1100

### CODE REVIEW

File: Api/Infrastructure/Options/SearchLimits.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿namespace HappyTravel.Edo.Api.Infrastructure.Options;
+
+public class SearchLimits
+{
+    public int HtIds { get; set; }
```
Comment (mingazhev) [inline Api/Infrastructure/Options/SearchLimits.cs:5]: This is not a list to name in plural.
`MacHtIdCount` or `MaxHtIdNumber` maybe

File: Api/Infrastructure/Options/SearchLimits.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    public int Rooms { get; set; }
```
Comment (mingazhev) [inline Api/Infrastructure/Options/SearchLimits.cs:6]: Same there, RoomCount, GuestCount

File: Api/Infrastructure/ServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -422,6 +422,7 @@ public static class ServiceCollectionExtensions
             services.Configure<PaymentProcessorOption>(configuration.GetSection("PaymentProcessor"));
             services.Configure<MarkupPolicyStorageOptions>(configuration.GetSection("MarkupPolicyStorageOptions"));
             services.Configure<DiscountStorageOptions>(configuration.GetSection("DiscountStorageOptions"));
+            services.Configure<SearchLimits>(configuration.GetSection("SearchLimits"));
```
Comment (mingazhev) [inline Api/Infrastructure/ServiceCollectionExtensions.cs:425]: Good, to get this from config

File: Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilitySearchService.cs [hunk 4]
```diff
@@ -44,9 +48,18 @@ public async Task<Result<Guid>> StartSearch(AvailabilityRequest request, AgentCo
...
+                return Result.Failure<Guid>($"{nameof(request.HtIds)} limit exceeded");
+            
             if (request.CheckInDate.Date < _dateTimeProvider.UtcToday())
                 return Result.Failure<Guid>("Check in date must not be in the past");
             
+            if (request.RoomDetails.Count > _searchLimits.CurrentValue.Rooms)
+                return Result.Failure<Guid>($"{nameof(request.RoomDetails)} limit exceeded");
+            
+            if (request.RoomDetails.Any(r => (r.AdultsNumber + r.ChildrenAges?.Count ?? 0) > _searchLimits.CurrentValue.Guests))
+                return Result.Failure<Guid>("Guests limit exceeded");
```
Comment (mingazhev) [inline Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilitySearchService.cs:61]: Does not this deserve a separate validator?

File: Api/Infrastructure/Options/SearchLimits.cs (hunk 1 above)
Comment (mingazhev) [inline Api/Infrastructure/Options/SearchLimits.cs:5]: I think HtIds limits needs to be configurable depending on the location htIds types - we have them in mappercontracts, if I'm not mistaken

Comment (DFSko) [reply Api/Infrastructure/Options/SearchLimits.cs]: Fixed

File: Api/Infrastructure/Options/SearchLimits.cs (hunk 2 above)
Comment (DFSko) [reply Api/Infrastructure/Options/SearchLimits.cs]: Fixed

File: Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilitySearchService.cs (hunk 4 above)
Comment (DFSko) [reply Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilitySearchService.cs]: Moved

File: Api/Infrastructure/Options/SearchLimits.cs (hunk 1 above)
Comment (DFSko) [reply Api/Infrastructure/Options/SearchLimits.cs]: Fixed

**Review by mingazhev — APPROVED** [review]

**Pull request merged**